src/cudamatrix/cu-allocator.h

```diff
@@ -335,6 +335,11 @@ class CuMemoryAllocator {
   // this is only locked by the '*Locking' versions of the functions (necessary only
   // in multi-threaded applications).
   std::mutex mutex_;
+
+  // Keep track of the memory usage from the cache to track the maximum memory used by
+  //   the application
+  int max_allocated_memory_;
```

Comment:
please make this size_t, instead of int.  If int was 32-bit, this might not always be enough.